data_source_obmcs_core_private_ip_test.go

```diff
@@ -19,26 +19,29 @@ type DatasourcePrivateIPTestSuite struct {
 	Provider     terraform.ResourceProvider
 	Providers    map[string]terraform.ResourceProvider
 	ResourceName string
-	List         *baremetal.ListPrivateIPs
 }
 
 func (s *DatasourcePrivateIPTestSuite) SetupTest() {
 	s.Client = testAccClient
 	s.Provider = testAccProvider
 	s.Providers = testAccProviders
-	s.Config = vnicConfig + `
-resource "oci_core_private_ip" "testPrivateIP" {
-	vnic_id = "${lookup(data.oci_core_vnic_attachments.vnics.vnic_attachments[0],"vnic_id")}"
-	ip_address = "10.0.1.23"
-}
-	`
-	s.Config += testProviderConfig()
-	s.ResourceName = "data.oci_core_private_ips.testPrivateIPs"
-}
+	s.Config = testProvider1() + testADs() + testVCN1() + testSubnet1() + testImage1() + testInstance1() + `	
```

Comment:
neat... would it make sense for each test resource to include its dependencies?

Reply:
That model gets complicated and breaks down pretty quickly when you need two things that both require the same upstream dependencies. I think aggregating functions that use the above could be a reasonable balance, example:
```
func testSubnetScenario() string {
  return testProvider1() + testADs() + testVCN1() + testSubnet1()
}

func testInstanceScenario() string {
  return testSubnetScenario() + testImage1() + testInstance1() 
}
```
But this can lead to its own complexity, packaged differently--what would the testInstanceScenario() look like if it requires 2 instances and 2 subnets?

Long term this is the beginning of being able to parameterize some of these functions.